MonumentAddons.cs

```diff
@@ -3631,6 +3607,249 @@ private interface IDeepCollection
             return false;
         }
 
+        private class MonumentHelper
+        {
+            public DungeonEntranceMapper _dungeonEntranceMapper;
+
+            private MonumentAddons _pluginInstance;
+            private Plugin _monumentFinder => _pluginInstance.MonumentFinder;
+           
+            public void OnServerInitialized(MonumentAddons pluginInstance) 
+            {
+                _pluginInstance = pluginInstance;
+                _dungeonEntranceMapper = new DungeonEntranceMapper();
```

Comment:
I suggest inlining this construction to where the field is defined since it takes no parameters.